include/sys/arc.h

```diff
@@ -154,6 +154,11 @@ typedef enum arc_flags
 	 */
 	ARC_FLAG_CACHED_ONLY		= 1 << 22,
 
+	/*
+	 * Don't instantiate an arc_buf_t for arc_read_done.
+	 */
+	ARC_FLAG_CALLBACK_ONLY		= 1 << 23,
```

Comment:
The name `CALLBACK_ONLY` doesn't immediately tell you what else could be done that we're excluding by "only" doing the callback (the comment explains it).  I was thinking that `ARC_FLAG_NO_BUF` might be more descriptive of a name.  i.e. "Do not instantiate/provide a buf when invoking the callback".  But I'm OK with the current name if you prefer it.